### PR TITLE
[do not integrate] Nano experiments

### PR DESCRIPTION
**Pull request by pruivo**

### CODE REVIEW

**Comment by pruivo** [comment]
run performance tests please

File: core/src/main/java/org/infinispan/container/EntryFactoryImpl.java [hunk 1]
```diff
@@ -209,6 +209,18 @@ private CacheEntry getFromContainer(Object key, boolean isOwner, boolean writeOp
...
 
+   private CacheEntry getFromContainerForRead(Object key, boolean isOwner) {
+      InternalCacheEntry ice = container.get(key);
+      if (trace) {
+         log.tracef("Retrieved from container %s", ice);
+      }
+      if (isOwner) {
+         return ice == null ? NullCacheEntry.getInstance() : ice;
+      } else {
+         return ice == null || !ice.isL1Entry() ? null : ice;
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/container/EntryFactoryImpl.java]: I could have read the code wrongly but this method is only invoked when it ```isOwner==true`` or ``isL1Enabled==true``(the condition is checked in ```wrapEntryForReading()```. In any case, the entry should be valid

**Comment by Holmistr** [comment]
@pruivo Due to changes yesterday, the PerfAck job wasn't triggered. I'm making some enhancements today, I will make sure to re-trigger it when done.

**Comment by pruivo** [comment]
Although the changes affects all configuration, the configuration that may benefit the most is ```library+replicated+reads+no-tx```

File: core/src/main/java/org/infinispan/container/EntryFactoryImpl.java (hunk 1 above)
Comment (danberindei) [inline core/src/main/java/org/infinispan/container/EntryFactoryImpl.java]: Right, I missed that we ignore non-L1 entries here. Then the only difference from getFromContainer() is that this method is skipping the if (writeOperation) check in innerGetFromContainer()?

**Comment by danberindei** [comment]
@pruivo There are lots of test failures.

File: core/src/main/java/org/infinispan/container/EntryFactoryImpl.java (hunk 1 above)
Comment (pruivo) [inline core/src/main/java/org/infinispan/container/EntryFactoryImpl.java]: yes

**Comment by Holmistr** [comment]
run performance tests please
library repl reads

**Comment by Holmistr** [comment]
@pruivo PerfAck again needs to be aborted because of higher priority testing.

**Comment by Holmistr** [comment]
run performance tests please
library repl reads

**Pull request closed without merging**